atomic_reactor/inner.py

```diff
@@ -245,11 +245,16 @@ class DockerBuildWorkflow(object):
     5. tag it
     6. push it to registries
     """
+    def __init__(self, build_json):
+        """
+        init wrapper, so we can process build_json intact as well
+        """
+        self.__do_init__(build_json, **build_json)
 
-    def __init__(self, source, image, prebuild_plugins=None, prepublish_plugins=None,
```

Comment:
A less intrusive approach would be to add an additional optional param to `__init__`: `raw_build_json`.
Initializing the workflow would simply change to: `DockerBuildWorkflow(raw_build_json=build_json, **build_json)`
This would avoid the need to `__do_init__` hack.